torat_client/rpcapi.go

```diff
@@ -71,9 +73,29 @@ func (a *API) LS(v shared.Void, r *shared.Dir) (err error) {
 	return
 }
 
-func (a *API) Ping(v shared.Void, r *string) error {
-	// TODO implement
-	*r = "Pong"
+func (a *API) Speedtest(v shared.Void, r *shared.Speedtest) error {
+
+	url := "https://api.ipify.org?format=text"
+	resp, _ := http.Get(url)
+	ip, _ := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	r.IP = string(ip)
+
+	user, _ := speedtest.FetchUserInfo()
```

Comment:
The user var should also contain the public ip, if so please remove the block above checking the public ip.

https://pkg.go.dev/github.com/showwin/speedtest-go@v1.1.1/speedtest#User

Reply:
This is a **great** catch. I thought it only returned the ip of the server tested.